custom_components/ha_hatch/light.py

```diff
@@ -61,17 +66,27 @@ def turn_on(self, **kwargs):
             # If 100 is sent to Abode, response is 99 causing an error
             brightness = round(kwargs[ATTR_BRIGHTNESS] * 100 / 255.0)
         else:
-            brightness = self._attr_brightness
+            if isinstance(self.rest_device, RestIot):
+                brightness = 0
+            else:
+                brightness = self._attr_brightness
         if ATTR_RGB_COLOR in kwargs:
             rgb = kwargs[ATTR_RGB_COLOR]
         else:
             rgb = self._attr_rgb_color
 
         _LOGGER.debug(f"turning on light to {rgb} with {brightness}")
-        self.rest_device.set_color(rgb[0], rgb[1], rgb[2], brightness)
+        if isinstance(self.rest_device, RestIot):
+            self.rest_device.set_color(rgb[0], rgb[1], rgb[2], 0, brightness, True)
+        else:
+            self.rest_device.set_color(rgb[0], rgb[1], rgb[2], brightness)
         if self.config_turn_on_light:
             _LOGGER.debug(f"auto turning on the hatch power switch for the light")
             self.rest_device.set_on(True)
 
     def turn_off(self):
-        self.rest_device.set_color(self.rest_device.red, self.rest_device.green, self.rest_device.blue, 0)
+        if isinstance(self.rest_device, RestIot):
+            _LOGGER.debug("turning off with rgbw = 0 and bright = 0")
+            self.rest_device.set_color(0, 0, 0, 0, 0, False)
```

Comment:
looks like you are having to modify the logic in all of the methods of HatchLightEntity, would be easier to maintain if we just make two classes, maybe HatchLightPlusEntity and HatchLightIotEntity

Reply:
it seems confusing to have 5 parameters that will never be needed if the 6th is, did you consider separating the set_color from turn_off_light instead? it doesn't match the plus either way because i didn't expose the w value and am curious how 0 brightness works differently in the iot device from the custom color id, looks like the gen2 api makes it harder to specify the light attributes without also having to set the sound (saw that weird remote thing too) I haven't worked with that api so I don't feel strongly either way.

Reply:
Yeah, this is a good idea.

Reply:
> looks like you are having to modify the logic in all of the methods of HatchLightEntity, would be easier to maintain if we just make two classes, maybe HatchLightPlusEntity and HatchLightIotEntity

Yeah this is a good idea.

Reply:
> it seems confusing to have 5 parameters that will never be needed if the 6th is, did you consider separating the set_color from turn_off_light instead? it doesn't match the plus either way because i didn't expose the w value and am curious how 0 brightness works differently in the iot device from the custom color id, looks like the gen2 api makes it harder to specify the light attributes without also having to set the sound (saw that weird remote thing too) I haven't worked with that api so I don't feel strongly either way.

Yeah they don't make it easy to change just one thing. Should be able to clean this up with the new entity like you suggested.

Reply:
Split the on/color change and turning off functions in the library so it should make this logic more clear.